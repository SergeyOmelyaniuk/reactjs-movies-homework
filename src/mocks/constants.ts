import { Category, Genre, Language } from '../types';

export const categories: Category[] = [
	{ id: 1, value: 'popular', title: 'Popular' },
	{ id: 2, value: 'top_rated', title: 'Top rated' },
	{ id: 3, value: 'upcoming', title: 'Upcoming' },
];

export const languages: Language[] = [
	{ title: 'en', value: 'en-US' },
	{ title: 'ru', value: 'ru-RU' },
];

export const genres: Genre[] = [
	{ id: 1, name: 'Genre 1' },
	{ id: 2, name: 'Genre 2' },
	{ id: 3, name: 'Genre 3' },
	{ id: 4, name: 'Genre 4' },
	{ id: 5, name: 'Genre 5' },
];
