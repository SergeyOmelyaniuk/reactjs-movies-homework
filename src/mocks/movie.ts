import { MovieDetails } from './../types';

export const movieDescription: MovieDetails = {
	poster_path: 'film_1.jpg',
	genres: [{ id: 1, name: 'first genre' }],
	title: 'Красотка на взводе',
	release_date: '2021-05-26',
	vote_average: 68,
	revenue: 120000,
	runtime: 120,
	overview:
		'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
};
