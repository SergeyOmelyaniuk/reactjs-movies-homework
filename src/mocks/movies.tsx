import { Movies } from '../types';

export const movies: Movies = {
	popular: [
		{
			poster_path: 'film_1.jpg',
			genre_ids: ['Genre 1', 'Genre 2'],
			id: 1,
			title: 'Красотка на взводе',
			release_date: '2021-05-26',
			genre: ['Боевик', 'Комедия'],
			img: 'film_1.jpg',
			vote_average: 68,
			popularity: 0.5,
			revenue: 120000,
			duration: '2:14',
			overview:
				'Вышибала Линди плохо умеет управлять собственным гневом и постоянно боится кого-то убить в приступе ярости. В качестве самоконтроля она носит специальный жилет, который бьёт её током и успокаивает, прежде чем она совершит необратимое. Но после смерти возлюбленного Линди вынуждена забыть о контроле и отправиться на поиски убийцы.',
			images: [
				'img_1.jpg',
				'img_2.jpg',
				'img_3.jpg',
				'img_4.jpg',
				'img_5.jpg',
				'img_6.jpg',
				'img_7.jpg',
				'img_8.jpg',
			],
			actors: [
				{
					name: 'Kate Beckinsale',
					birthday: '1959-04-15',
					placeBirth: 'Brest, Belarus',
					biography: 'Text',
					photo: 'photo_1.jpg',
					photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
					character: 'Lindy',
				},
				{
					name: 'Stanley Tucci',
					birthday: '1959-04-15',
					placeBirth: 'Brest, Belarus',
					biography: 'Text',
					photo: 'photo_2.jpg',
					photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
					character: 'Dr. Ivan Munchin',
				},
				{
					name: 'Bobby Cannavale',
					birthday: '1959-04-15',
					placeBirth: 'Brest, Belarus',
					biography: 'Text',
					photo: 'photo_3.jpg',
					photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
					character: 'Detective Vicars',
				},
				{
					name: 'Jai Courtney',
					birthday: '1959-04-15',
					placeBirth: 'Brest, Belarus',
					biography: 'Text',
					photo: 'photo_4.jpg',
					photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
					character: 'Justin',
				},
				{
					name: 'Laverne Cox',
					birthday: '1959-04-15',
					placeBirth: 'Brest, Belarus',
					biography: 'Text',
					photo: 'photo_5.jpg',
					photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
					character: 'Detective Nevin',
				},
				{
					name: 'David Bradley',
					birthday: '1959-04-15',
					placeBirth: 'Brest, Belarus',
					biography: 'Text',
					photo: 'photo_6.jpg',
					photos: ['photo_1.jpg', 'photo_2.jpg', 'photo_3.jpg', 'photo_4.jpg'],
					character: 'Gareth Fizel',
				},
			],
		},
		{
			poster_path: 'film_2.jpg',
			genre_ids: ['Genre 1', 'Genre 2'],
			id: 2,
			title: 'Сокровище Счастливчика',
			release_date: '2021-05-26',
			genre: ['Боевик', 'Комедия', 'Драма'],
			img: 'film_2.jpg',
			vote_average: 72,
			popularity: 0.5,
			revenue: 230000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 3,
			poster_path: 'film_3.jpg',
			genre_ids: ['Genre 1'],
			title: 'Сталинград',
			release_date: '2021-05-26',
			genre: ['Боевик', 'Комедия'],
			img: 'film_3.jpg',
			vote_average: 76,
			popularity: 0.5,
			revenue: 240000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 4,
			poster_path: 'film42.jpg',
			genre_ids: ['Genre 4'],
			title: 'Люди в черном: Интернэшнл',
			release_date: '2021-05-26',
			genre: ['Боевик', 'Комедия'],
			img: 'film_4.jpg',
			vote_average: 69,
			popularity: 0.5,
			revenue: 136000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 5,
			poster_path: 'film_5.jpg',
			genre_ids: ['Genre 1'],
			title: 'Значит, война',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_5.jpg',
			vote_average: 96,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 6,
			poster_path: 'film_6.jpg',
			genre_ids: ['Genre 1'],
			title: 'Приключения желтого чемоданчика',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_6.jpg',
			vote_average: 77,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			poster_path: 'film_7.jpg',
			genre_ids: ['Genre 1'],
			id: 7,
			title: 'Поезд в Пусан',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_7.jpg',
			vote_average: 79,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 8,
			poster_path: 'film_8.jpg',
			genre_ids: ['Genre 3'],
			title: 'Никто не выжил',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_8.jpg',
			vote_average: 79,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 9,
			poster_path: 'film_9.jpg',
			genre_ids: ['Genre 1'],
			title: 'Ведьма',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_9.jpg',
			vote_average: 53,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 10,
			poster_path: 'film_10.jpg',
			genre_ids: ['Genre 1'],
			title: 'Звёздные войны: Пробуждение силы',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_10.jpg',
			vote_average: 53,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 11,
			poster_path: 'film_11.jpg',
			genre_ids: ['Genre 1'],
			title: 'Марсианин',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_11.jpg',
			vote_average: 58,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 12,
			poster_path: 'film_12jpg',
			genre_ids: ['Genre 3'],
			title: 'Республика Z',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_12.jpg',
			vote_average: 100,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 13,
			poster_path: 'film_13.jpg',
			genre_ids: ['Genre 1'],
			title: 'Планета 51',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_13.jpg',
			vote_average: 90,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 14,
			poster_path: 'film_14.jpg',
			genre_ids: ['Genre 1'],
			title: 'Красные туфельки и семь гномов',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_14.jpg',
			vote_average: 86,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 15,
			poster_path: 'film_15.jpg',
			genre_ids: ['Genre 1'],
			title: 'Капитошка',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_15.jpg',
			vote_average: 92,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 16,
			poster_path: 'film_16.jpg',
			genre_ids: ['Genre 1'],
			title: 'Война будущего',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_16.jpg',
			vote_average: 58,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 17,
			poster_path: 'film_17.jpg',
			genre_ids: ['Genre 5'],
			title: 'Чернобыль',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_17.jpg',
			vote_average: 86,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 18,
			poster_path: 'film_18.jpg',
			genre_ids: ['Genre 3'],
			title: 'Странный дом',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_18.jpg',
			vote_average: 81,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 19,
			poster_path: 'film_19.jpg',
			genre_ids: ['Genre 1'],
			title: 'Дальний космос',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_19.jpg',
			vote_average: 79,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 20,
			poster_path: 'film_20.jpg',
			genre_ids: ['Genre 1'],
			title: 'Мандалорец',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_20.jpg',
			vote_average: 20,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
	],
	top_rated: [
		{
			id: 21,
			poster_path: 'film_21.jpg',
			genre_ids: ['Genre 1'],
			title: 'Интерстеллар',
			release_date: '2021-05-26',
			genre: ['Драма'],
			img: 'film_21.jpg',
			vote_average: 56,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 22,
			poster_path: 'film_22.jpg',
			genre_ids: ['Genre 2'],
			title: 'Безумный Макс: Дорога ярости',
			release_date: '2021-05-26',
			genre: ['Драма'],
			img: 'film_22.jpg',
			vote_average: 89,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 23,
			poster_path: 'film_23.jpg',
			genre_ids: ['Genre 2'],
			title: 'Гонка',
			release_date: '2021-05-26',
			genre: ['Драма'],
			img: 'film_23.jpg',
			vote_average: 56,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 24,
			poster_path: 'film_24.jpg',
			genre_ids: ['Genre 2'],
			title: 'Джокер',
			release_date: '2021-05-26',
			genre: ['Драма'],
			img: 'film_24.jpg',
			vote_average: 24,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 25,
			poster_path: 'film_25.jpg',
			genre_ids: ['Genre 2'],
			title: 'Отступники',
			release_date: '2021-05-26',
			genre: ['Драма'],
			img: 'film_25.jpg',
			vote_average: 79,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 26,
			poster_path: 'film_26.jpg',
			genre_ids: ['Genre 2'],
			title: 'Мстители: Финал',
			release_date: '2021-05-26',
			genre: ['Драма'],
			img: 'film_26.jpg',
			vote_average: 80,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 27,
			poster_path: 'film_27.jpg',
			genre_ids: ['Genre 2'],
			title: '1917',
			release_date: '2021-05-26',
			genre: ['Драма'],
			img: 'film_27.jpg',
			vote_average: 76,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 28,
			poster_path: 'film_28.jpg',
			genre_ids: ['Genre 2'],
			title: 'Тьма',
			release_date: '2021-05-26',
			genre: ['Драма'],
			img: 'film_28.jpg',
			vote_average: 55,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 29,
			poster_path: 'film_29.jpg',
			genre_ids: ['Genre 2'],
			title: 'Остров проклятых',
			release_date: '2021-05-26',
			genre: ['Драма'],
			img: 'film_29.jpg',
			vote_average: 11,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 30,
			poster_path: 'film_30.jpg',
			genre_ids: ['Genre 3'],
			title: 'Дикие истории',
			release_date: '2021-05-26',
			genre: ['Драма'],
			img: 'film_30.jpg',
			vote_average: 46,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
	],
	upcoming: [
		{
			id: 31,
			poster_path: 'film_31.jpg',
			genre_ids: ['Genre 3'],
			title: 'Бегущий по лезвию 2049',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_31.jpg',
			vote_average: 68,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 32,
			poster_path: 'film_32.jpg',
			genre_ids: ['Genre 3'],
			title: 'Клаустрофобы 2: Лига выживших',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_32.jpg',
			vote_average: 97,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 33,
			poster_path: 'film_33.jpg',
			genre_ids: ['Genre 3'],
			title: 'В петле времени',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_33.jpg',
			vote_average: 89,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 34,
			poster_path: 'film_34.jpg',
			genre_ids: ['Genre 3'],
			title: 'Дальний космос',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_34.jpg',
			vote_average: 89,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
		{
			id: 35,
			poster_path: 'film_35.jpg',
			genre_ids: ['Genre 3'],
			title: 'Как приручить дракона 3',
			release_date: '2021-05-26',
			genre: ['Боевик'],
			img: 'film_35.jpg',
			vote_average: 55,
			popularity: 0.5,
			revenue: 226000,
			duration: '2:14',
			overview: '',
			images: [],
			actors: [],
		},
	],
};
