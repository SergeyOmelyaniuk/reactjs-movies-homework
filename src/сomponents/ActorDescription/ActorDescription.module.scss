.actorDescription {
	display: flex;
}

.wrapPhotoActor {
	max-width: 425px;
	width: 100%;
	height: 640px;
	margin-right: 40px;
	flex-shrink: 0;
}

.photoActor {
	height: 100%;
	max-width: 100%;
	display: block;
	border-radius: 5px;
}

.infoActor {
	font-weight: 300;
	font-size: 16px;
}

.nameActor {
	font-size: 46px;
	margin-bottom: 50px;
}

.textBlock {
	margin-bottom: 50px;
}

.title {
	display: block;
	margin-bottom: 10px;
}

.date {
	font-size: 20px;
}

.placeBirthActor {
	font-size: 20px;
}

.text {
	line-height: 1.5;
	font-size: 20px;
}

.titlePhotosActor {
	font-size: 28px;
	display: block;
	margin-bottom: 30px;
}

.wrapPhotosActor {
	display: flex;
}

.wrapPhotoSmallActor {
	max-width: 150px;
	width: 100%;
	height: 230px;
	margin-right: 40px;
}

.photoSmallActor {
	height: 100%;
	max-width: 100%;
	display: block;
	border-radius: 5px;
}
