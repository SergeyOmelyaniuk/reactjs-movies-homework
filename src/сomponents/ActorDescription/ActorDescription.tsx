import React from 'react';
import styles from './ActorDescription.module.scss';
import { Actor } from '../../types';
import TextBlock from '../TextBlock';

interface ActorDescriptionProps {
	actor: Actor;
}

const ActorDescription = ({ actor }: ActorDescriptionProps) => (
	<div className={styles.actorDescription}>
		<div className={styles.wrapPhotoActor}>
			<img
				className={styles.photoActor}
				src={require(`../../mocks/images/actors/${actor.photo}`).default}
				alt=''
			/>
		</div>
		<div className={styles.infoActor}>
			<h2 className={styles.nameActor}>{actor.name}</h2>
			<TextBlock title='Birthday' value={actor.birthday} />
			<TextBlock title='Place of birth' value={actor.placeBirth} />
			<TextBlock title='Biography' value={actor.biography} />
			<div className={styles.photosActor}>
				<span className={styles.titlePhotosActor}>Photos</span>
				<div className={styles.wrapPhotosActor}>
					{actor.photos.map((photo) => {
						return (
							<div key={photo} className={styles.wrapPhotoSmallActor}>
								<img
									className={styles.photoSmallActor}
									src={
										require(`../../mocks/images/additional_photos/${photo}`)
											.default
									}
									alt=''
								/>
							</div>
						);
					})}
				</div>
			</div>
		</div>
	</div>
);

export default ActorDescription;
