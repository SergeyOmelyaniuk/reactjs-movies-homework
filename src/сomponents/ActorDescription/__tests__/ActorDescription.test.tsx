import React from 'react';
import { render } from '@testing-library/react';
import ActorDescription from '..';
import { movies } from '../../../mocks/movies';

describe('ActorDescription', () => {
	test('should render', () => {
		const { container } = render(
			<ActorDescription actor={movies.popular[0].actors[0]} />
		);

		expect(container).toMatchSnapshot();
	});
});
