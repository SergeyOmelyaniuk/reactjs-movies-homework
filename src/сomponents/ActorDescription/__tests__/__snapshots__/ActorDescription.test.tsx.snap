// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorDescription should render 1`] = `
<div>
  <div
    class="actorDescription"
  >
    <div
      class="wrapPhotoActor"
    >
      <img
        alt=""
        class="photoActor"
      />
    </div>
    <div
      class="infoActor"
    >
      <h2
        class="nameActor"
      >
        Kate Beckinsale
      </h2>
      <div
        class="textBlock"
      >
        <span
          class="title"
        >
          Birthday
          :
        </span>
        <p
          class="text"
        >
          1959-04-15
        </p>
      </div>
      <div
        class="textBlock"
      >
        <span
          class="title"
        >
          Place of birth
          :
        </span>
        <p
          class="text"
        >
          Brest, Belarus
        </p>
      </div>
      <div
        class="textBlock"
      >
        <span
          class="title"
        >
          Biography
          :
        </span>
        <p
          class="text"
        >
          Text
        </p>
      </div>
      <div
        class="photosActor"
      >
        <span
          class="titlePhotosActor"
        >
          Photos
        </span>
        <div
          class="wrapPhotosActor"
        >
          <div
            class="wrapPhotoSmallActor"
          >
            <img
              alt=""
              class="photoSmallActor"
            />
          </div>
          <div
            class="wrapPhotoSmallActor"
          >
            <img
              alt=""
              class="photoSmallActor"
            />
          </div>
          <div
            class="wrapPhotoSmallActor"
          >
            <img
              alt=""
              class="photoSmallActor"
            />
          </div>
          <div
            class="wrapPhotoSmallActor"
          >
            <img
              alt=""
              class="photoSmallActor"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
